### ISSUE TITLE
`useless_let_if_seq` FN when `if` is in the last expr of block

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#16062]
### Summary

`useless_let_if_seq` FN when `if` is in the last expr of block

### Lint Name

useless_let_if_seq

### Reproducer

I tried this code:

```rust
fn test(bar: fn() -> bool) {
    let foo;
    if bar() {
        foo = 42;
    } else {
        foo = 0;
    }
}
```

I expected to see this happen:
```
 --> src/main.rs:5:5
  |
5 |     let foo;
  |     ^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_late_init
  = note: `#[warn(clippy::needless_late_init)]` on by default
help: move the declaration `foo` here and remove the assignments from the branches
  |
5 ~     
6 ~     let foo = if bar() {
7 ~         42
8 |     } else {
9 ~         0
```

Instead, this happened:
Nothing

### Version

```text
rustc 1.93.0-nightly (d5419f1e9 2025-10-30)
binary: rustc
commit-hash: d5419f1e97b90741d51841f800d3c697c662567d
commit-date: 2025-10-30
host: x86_64-unknown-linux-gnu
release: 1.93.0-nightly
LLVM version: 21.1.3
```
